src/lsstseries/utils/column_mapper/column_mapper.py

```diff
@@ -0,0 +1,200 @@
+class ColumnMapper:
+    """Maps columns from a given dataset into known ensemble column"""
+
+    def __init__(
+        self,
+        id_col=None,
+        time_col=None,
+        flux_col=None,
+        err_col=None,
+        band_col=None,
+        provenance_col=None,
+        nobs_total_col=None,
+        nobs_band_cols=None,
+    ):
+        """
+
+        Parameters
+        ----------
+        id_col: 'str', optional
+            Identifies which column contains the Object IDs
+        time_col: 'str', optional
+            Identifies which column contains the time information
+        flux_col: 'str', optional
+            Identifies which column contains the flux/magnitude information
+        err_col: 'str', optional
+            Identifies which column contains the flux/mag error information
+        band_col: 'str', optional
+            Identifies which column contains the band information
+        nobs_col: list of 'str', optional
+            Identifies which columns contain number of observations for each
+            band, if available in the input object file
+        nobs_tot_col: 'str', optional
+            Identifies which column contains the total number of observations,
+            if available in the input object file
+        provenance_col: 'str', optional
+            Identifies which column contains the provenance information, if
+            None the provenance column is generated.
+
+        Returns
+        -------
+        ColumnMapper object
+        """
+
+        self.map = {
+            "id_col": id_col,
+            "time_col": time_col,
+            "flux_col": flux_col,
+            "err_col": err_col,
+            "band_col": band_col,
+            "provenance_col": provenance_col,
+            "nobs_total_col": nobs_total_col,
+            "nobs_band_cols": nobs_band_cols,
+        }
+
+        # Specifies which column mappings must be set for the Ensemble
+        self.required = {
+            "id_col": True,
+            "flux_col": True,
+            "time_col": True,
+            "err_col": True,
+            "band_col": True,
+            "provenance_col": False,
+            "nobs_total_col": False,
+            "nobs_band_cols": False,
+        }
+
+        self.known_maps = {"ZTF": ZTFColumnMapper}
+
+    def _set_known_map(self):
+        """Must be defined in a known map class"""
+        raise NotImplementedError
+
+    @staticmethod
+    def map_id() -> str:
+        return None
+
+    def use_known_map(self, map_id):
+        """Use a known mapping scheme
+
+        Parameters
+        ----------
+        map_id: 'str'
+            Identifies which mapping scheme to use
+
+        Returns
+        -------
+        A ColumnMapper subclass object dependent on the map_id provided, for example
+        ZTFColumnMapper in the case of "ZTF"
+
+        """
+        if map_id in self.known_maps.keys():
+            if map_id == "ZTF":
```

Comment:
I don't think you need the conditional check here. Can this be reduced to:
```
if map_id in self.known_maps:
    return self.known_maps[map_id]()._set_known_map()
else
    ...
```

Perhaps it's less clear this way ☝️ , if so, feel free to leave as is. 
Either way, it would be nice to avoid having to extend the conditional for each new `map_id`.